org.osgi.test.cases.cm/src/org/osgi/test/cases/cm/junit/CMControl.java

```diff
@@ -311,17 +361,19 @@ private void printoutPermissions() {
 	}
 
 	private void setBundlePermission(Bundle b, List<PermissionInfo> list) {
-		if (permAdmin == null)
+		if (permAdminA.getService() == null)
```

Comment:
You should put `permAdminA.getService()` into a variable and then test for `null` and use. Calling `permAdminA.getService()` twice could give different results.